src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/CallSiteFactory.cs

```diff
@@ -523,7 +524,7 @@ private ServiceCallSite TryCreateOpenGeneric(ServiceDescriptor descriptor, Type
         }
 
 
-        public void Add(Type type, ServiceCallSite serviceCallSite)
+        public void Add(Type? type, ServiceCallSite serviceCallSite)
```

Comment:
Who calls this with a `null` type?

Reply:
Nobody. It's just there were no null-checks and I assumed `ServiceCacheKey` accepts null `type` because of the `Empty`, so annotated `Add` like that. 

Reverted this change.